### PR TITLE
Add integration test

### PR DESCRIPTION
**Pull request by lojzatran**
I added one integration test for creating template order from a checkout order. It is failing because there is no implementation in this PR.

### CODE REVIEW

File: test/resources/payment-draft.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "amountPlanned": {
+    "type": "centPrecision",
+    "currencyCode": "EUR",
+    "centAmount": 795,
+    "fractionDigits": 2
+  },
+  "paymentMethodInfo": {
+    "paymentInterface": "ctp-adyen-integration",
```
Comment (ahmetoz) [inline test/resources/payment-draft.json:9]: maybe we should use more dummy examples ? 😄

File: test/integration/runner.spec.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+      .execute()
+    expect(checkoutOrderUpdated.custom.fields.isSubscriptionProcessed).to.be
+      .true
+  })
+
+  function assertPayments(checkoutOrder, templateOrders) {
+    const checkoutOrderPayment = _.cloneDeep(
+      checkoutOrder.paymentInfo.payments[0].obj
+    )
+    delete checkoutOrderPayment.id
```
Comment (ahmetoz) [inline test/integration/runner.spec.js:95]: is there another way to avoid deletes but assert all fields ? I could not reason much if those deletes are the right way to go ? what was the main idea behind that type of testing ?

File: test/integration/runner.spec.js [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    assertCustomFieldsEqual(checkoutOrder, templateOrders)
+    assertLineItemsEqual(checkoutOrder, templateOrders)
+    assertTemplateOrderCustomFields(checkoutOrder, templateOrders)
+    assertPayments(checkoutOrder, templateOrders)
+    const { body: checkoutOrderUpdated } = await apiRoot
+      .orders()
+      .withId({ ID: checkoutOrder.id })
+      .get()
+      .execute()
+    expect(checkoutOrderUpdated.custom.fields.isSubscriptionProcessed).to.be
```
Comment (ahmetoz) [inline test/integration/runner.spec.js:56]: maybe we could also assert if the state=Active ?

**Review by ahmetoz — COMMENTED** [review]
are we planning to write IT for the edge cases ? such as skipping some orders, counting retries etc.

File: test/resources/payment-draft.json (hunk 1 above)
Comment (lojzatran) [reply test/resources/payment-draft.json:9]: It's dummy, but at the same time it is a product placement for our great [adyen-integration project](https://github.com/commercetools/commercetools-adyen-integration)

File: test/integration/runner.spec.js (hunk 2 above)
Comment (lojzatran) [reply test/integration/runner.spec.js:95]: I don't know, this seems to be easiest way to compare if 2 objects are equal. I delete because IDs are obviously not equal, same applies for timestamps etc. Otherwise I would need to write every field I want to compare.

File: test/integration/runner.spec.js (hunk 3 above)
Comment (lojzatran) [reply test/integration/runner.spec.js:56]: https://github.com/commercetools/commercetools-subscriptions/pull/4/commits/fefc278402c36902cdaf076c1ecb028293ae5c7f#diff-5c3fb411d68af19e71866dc939ebee5f3d60be0eac74b6f40064480dbb7db99dR74

**Comment by lojzatran** [comment]
> are we planning to write IT for the edge cases ? such as skipping some orders, counting retries etc.

Not for now. It's too much for IT. Maybe we could do more unit tests, but not in this phase.

**Review by ahmetoz — APPROVED** [review]
